test/e2e/test/e2e.test.ts

```diff
@@ -37,22 +42,30 @@ beforeAll(async () => {
   const conf: RemoteOptions = {
     logLevel: 'info',
     port: 4723,
-    capabilities: undefined,
+    capabilities: {},
   };
 
+  if (process.env.APPIUM_APP === undefined) {
+    throw new Error('APPIUM_APP environment variable must be set');
+  }
+  if (process.env.PLATFORM === 'ios' &&
+      process.env.APPIUM_DERIVED_DATA === undefined) {
+    throw new Error('APPIUM_DERIVED_DATA environment variable must be set');
+  }
+
   if (process.env.PLATFORM === 'android') {
     conf.capabilities = {
       platformName: 'Android',
       'appium:automationName': 'UIAutomator2',
-      'appium:app': './android/app/build/outputs/apk/release/app-release.apk',
+      'appium:app': process.env.APPIUM_APP,
     };
   } else {
     conf.capabilities = {
       platformName: 'iOS',
       'appium:automationName': 'XCUITest',
-      'appium:app':
-        './ios/DerivedData/Build/Products/Release-iphonesimulator/sample.app',
-      'appium:derivedDataPath': path.resolve('./ios/DerivedData'),
+      'appium:app': process.env.APPIUM_APP,
+      // DerivedData of the WebDriverRunner Xcode project.
```

Comment:
Don't we need this anymore?

Reply:
We do need to set the app, but now it's done by ENV so it can be dynamic. For all the apps we are testing (different paths for different RN versions.)